src/probnum/filtsmooth/timeseriesposterior.py

```diff
@@ -36,9 +36,30 @@ class TimeSeriesPosterior(abc.ABC):
         Posterior random variables.
     """
 
-    def __init__(self, locations: np.ndarray, states: np.ndarray) -> None:
-        self.locations = np.asarray(locations)
-        self.states = _randomvariablelist._RandomVariableList(states)
+    def __init__(
+        self,
+        locations: Optional[Iterable[FloatArgType]] = None,
+        states: Optional[Iterable[randvars.RandomVariable]] = None,
+    ) -> None:
+        self._locations = list(locations) if locations is not None else []
+        self._states = list(states) if states is not None else []
+
+    def append(
+        self,
+        location: FloatArgType,
+        state: randvars.RandomVariable,
+    ) -> None:
+        # TODO: Type-checks here?
+        self._locations.append(location)
+        self._states.append(state)
+
+    @property
+    def locations(self):
+        return np.asarray(self._locations)
+
+    @property
+    def states(self):
+        return _randomvariablelist._RandomVariableList(self._states)
```

Comment:
this accesses a private class.

Reply:
see the "hence the underscore" bit in #388

Reply:
thanks